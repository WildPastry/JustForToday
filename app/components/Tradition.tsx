import { ITradition } from '../types/data.types';
import { StyleSheet } from 'react-native';
import { Text } from './Themed';

const TraditionItem: React.FC<ITradition> = (
  tradtion: ITradition
): JSX.Element => {
  return (
    // Tradition
    <Text style={styles.text}>
      #{tradtion.id}: {tradtion.tradition}
    </Text>
  );
};

const styles = StyleSheet.create({
  text: {
    textAlign: 'left',
    lineHeight: 20,
    marginBottom: 10
  }
});

export default TraditionItem;
