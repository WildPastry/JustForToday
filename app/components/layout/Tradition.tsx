import { FontBold, FontRegular } from '../styles/StyledText';
import { ITradition } from '../../types/data.types';
import { StyleSheet } from 'react-native';
import { View } from '../../components/styles/Themed';
import globlStyles from './../../constants/styles';

const TraditionItem: React.FC<ITradition> = (
  props: ITradition
): JSX.Element => {
  return (
    // Tradition
    <View style={styles.stepContainer}>
      <FontBold style={globlStyles.number}>{props.id}</FontBold>
      <FontRegular style={globlStyles.body}>{props.tradition}</FontRegular>
    </View>
  );
};

const styles = StyleSheet.create({
  stepContainer: {
    flexDirection: 'row',
    marginBottom: 20,
    width: '90%'
  }
});

export default TraditionItem;
