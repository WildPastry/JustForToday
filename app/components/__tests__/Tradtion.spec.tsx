import { Provider } from 'react-redux';
import Tradition from '../Tradition';
import { makeStore } from '../../redux/store';
import renderer from 'react-test-renderer';

describe('<Tradition />', () => {
  it('renders correctly', () => {
    const store = makeStore();
    const tree = renderer
      .create(
        <Provider store={store}>
          <Tradition id={''} tradition={''} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
