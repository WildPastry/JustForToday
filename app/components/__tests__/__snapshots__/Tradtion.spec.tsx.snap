// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Tradition /> renders correctly 1`] = `
<Text
  style={
    [
      {
        "color": "#000",
      },
      {
        "flexDirection": "row",
        "marginTop": 20,
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "fontWeight": "200",
            "marginRight": 10,
          },
          {
            "fontSize": 24,
            "lineHeight": 34,
          },
        ],
      ]
    }
  >
      
  </Text>
  <Text
    style={
      [
        {
          "color": "#000",
        },
        [
          {
            "textAlign": "left",
          },
          {
            "fontSize": 24,
            "lineHeight": 34,
          },
        ],
      ]
    }
  />
</Text>
`;
